cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go

```diff
@@ -142,26 +160,105 @@ var _ = Describe("[sriov] NUMA node alignment", Ordered, func() {
 		By("Create server Pod and run E2E ICMP validation")
 		validateE2EICMPTraffic(pod, fmt.Sprintf(`[{"name": "test-numa-1-exclude-topology-true-network","ips":["192.0.2.250/24"]}]`))
 	})
-})
 
-func findDeviceOnNUMANode(node *corev1.Node, devices []*sriovv1.InterfaceExt, numaNode string) (*sriovv1.InterfaceExt, error) {
-	for _, device := range devices {
-		out, err := nodes.ExecCommandOnNode([]string{
-			"cat",
-			filepath.Clean(filepath.Join("/sys/class/net/", device.Name, "/device/numa_node")),
-		}, node)
-		if err != nil {
-			klog.Warningf("can't get device [%s] NUMA node: out(%s) err(%s)", device.Name, string(out), err.Error())
-			continue
+	It("Validate the creation of a pod with excludeTopology set to True and an SRIOV interface in a different NUMA node "+
+		"than the pod", func() {
+		pod := pods.DefinePod(sriovnamespaces.Test)
+		pods.RedefineWithGuaranteedQoS(pod, "2", "500Mi")
+		pod = pods.RedefinePodWithNetwork(pod, "test-numa-0-exclude-topology-true-network")
+
+		pod, err := client.Client.Pods(sriovnamespaces.Test).
+			Create(context.Background(), pod, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func(g Gomega) {
+			actualPod, err := client.Client.Pods(sriovnamespaces.Test).Get(context.Background(), pod.Name, metav1.GetOptions{})
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(actualPod.Status.Phase).To(Equal(corev1.PodRunning))
+			g.Expect(actualPod.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed))
+		}, 30*time.Second, 1*time.Second).Should(Succeed())
+
+		By("Create server Pod and run E2E ICMP validation")
+		validateE2EICMPTraffic(pod, fmt.Sprintf(`[{"name": "test-numa-0-exclude-topology-true-network","ips":["192.0.2.250/24"]}]`))
+	})
+
+	It("Validate the creation of a pod with two sriovnetworknodepolicies one with excludeTopology False and the "+
+		"second true each interface is in different NUMA as the pod", func() {
+
+		if len(numa0DeviceList) < 2 {
+			testSkip := "There are not enough Interfaces in NUMA Node 0 to complete this test"
+			Skip(testSkip)
 		}
 
-		if out == numaNode {
-			return device, nil
+		pod := pods.DefinePod(sriovnamespaces.Test)
+		pods.RedefineWithGuaranteedQoS(pod, "2", "500Mi")
+		pod = pods.RedefinePodWithNetwork(pod, "test-numa-0-nic1-exclude-topology-true-network, "+
+			"test-numa-0-nic2-exclude-topology-false-network")
+
+		pod, err := client.Client.Pods(sriovnamespaces.Test).
+			Create(context.Background(), pod, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func(g Gomega) {
+			actualPod, err := client.Client.Pods(sriovnamespaces.Test).Get(context.Background(), pod.Name, metav1.GetOptions{})
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(actualPod.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed))
+			g.Expect(actualPod.Status.Phase).To(Equal(corev1.PodFailed))
+			g.Expect(actualPod.Status.Message).To(ContainSubstring("Resources cannot be allocated with Topology locality"))
```

Comment:
Please, assert on `Status.Reason`

```suggestion
			g.Expect(actualPod.Status.Reason).To(Equal("TopologyAffinityError"))

```

Reply:
changed